### PR TITLE
Allow access tokens to be decoded

### LINKED ISSUE #570: ID tokens cannot be decoded because verification fails
### SDK Version

8.0.0

### PHP Version

PHP 8.0

### Composer Version

2.x

### What happened?

I have an ID token and when I call `$token = $auth0->decode(token: $jwt)` I get the following error:

```
Authorized Party (azp) claim mismatch in the ID token; expected "care-ops-backend", found "9NwyNHRcD2bNGKz25RwetasltUUchC9A"  
```

This is because of [this validation][1] being misconfigured to use `$tokenAudience` when it should be using `$clientId`. This would be solved by changing `->authorizedParty($tokenAudience)` to `->authorizedParty([$this->configuration->getClientId()])` in the line given (L160).

Additionally, if I remove `audience` from my configuration, I get a different error:

```
Audience (aud) claim mismatch in the token; expected "9NwyNHRcD2bNGKz25RwetasltUUchC9A", found "care-ops-backend, https://roundingwell-care-team.auth0.com/userinfo"
```

This is because the [default audience][2] is misconfigured. As far as I can tell, the only possible default audience is for ID tokens only, and that default would be:

```php
if (count($tokenAudience) === 0 && $this->type === self::TYPE_ID_TOKEN) {
    $tokenAudience[] = $this->configuration->formatDomain() . '/userinfo';
}
```

[1]: https://github.com/auth0/auth0-PHP/blob/8.0.0/src/Token.php#L160
[2]: https://github.com/auth0/auth0-PHP/blob/8.0.0/src/Token.php#L143-L146

### How can we reproduce this issue?

Attempt to call `$auth0->decode($jwt)` with any `$jwt` that is an ID token.

### Additional context

It also appears that [the documentation for ID token validation][1] is incorrect, because it states:

> The audience value for the token must match the client ID of the application as defined in your Application's Settings in the Client ID field.

The reality is that the `azp`, **NOT** `aud`, will be set to the application's Client ID value.

[1]: https://auth0.com/docs/security/tokens/id-tokens/validate-id-tokens

### PR DESCRIPTION
**Pull request by shadowhand**
### References

Refs #570 

### Contributor Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

File: tests/Utilities/TokenGenerator.php [hunk 1]
```diff
@@ -37,12 +37,12 @@ protected static function getIdTokenClaims(
         $defaults = [
             'sub' => '__test_sub__',
             'iss' => 'https://__test_domain__/',
-            'aud' => '__test_client_id__',
+            'aud' => 'https://__test_domain__/userinfo',
```
Comment (shadowhand) [inline tests/Utilities/TokenGenerator.php:40]: As per the issue, the Client ID is never used for ID tokens. However, all ID tokens contain `$DOMAIN/userinfo` and `$applicationName` in `aud`.

File: tests/Utilities/TokenGenerator.php [hunk 2]
```diff
@@ -37,12 +37,12 @@ protected static function getIdTokenClaims(
...
             'nonce' => '__test_nonce__',
             'auth_time' => time() - 100,
             'exp' => time() + 1000,
             'iat' => time() - 1000,
-            'azp' => '__test_azp__'
+            'azp' => '__test_client_id___'
```
Comment (shadowhand) [inline tests/Utilities/TokenGenerator.php:45]: As per the issue, ID tokens always use the client ID as the authorized party.

File: src/Auth0.php [hunk 3]
```diff
@@ -246,28 +246,28 @@ public function decode(
         ?string $tokenNonce = null,
         ?int $tokenMaxAge = null,
         ?int $tokenLeeway = null,
-        ?int $tokenNow = null
+        ?int $tokenNow = null,
+        ?int $tokenType = null
```
Comment (shadowhand) [inline src/Auth0.php:250]: @evansims updated this PR to add the missing `$tokenType` parameter for decoding.

**Comment by evansims** [comment]
Addressed via the linked issue 👍 Thanks for the contribution!

**Comment by shadowhand** [comment]
@evansims please reopen, I just updated the PR.

**Comment by evansims** [comment]
Whoops, that's timing! Thanks @shadowhand

**Review by evansims — APPROVED** [review]
This is a great improvement, thanks for all the good conversation around this, and of course your contribution with the PR

**Comment by shadowhand** [comment]
Thank you for your patience in tracking down the root of my misunderstanding!

**Pull request merged**